Objects/call.c

```diff
@@ -336,14 +361,27 @@ _PyObject_Call(PyThreadState *tstate, PyObject *callable,
             return NULL;
         }
 
+        /* Adhere to documented equivalence:
+           callable(*args, **kwargs) == PyObject_Call(callable, args, kwargs)
+
+           The dict must be copied to ensure the caller's copy isn't modified.
+           Not needed when following vectorcall paths that don't use the dict.
+           */
+        kwcopy = _new_dict_if_nonempty(kwargs);
```

Comment:
`kwcopy` ref leak if `_Py_EnterRecursiveCall` disallows the call.

Reply:
Good catch. Fixing.

Reply:
On checking local history, I'd originally copied and released inside the area protected by `_Py_EnterRecursiveCall`, which made it unnecessary. I changed my mind (those guards should really only guard the call itself, not whatever arbitrary code might get executed in the process of merging from a `dict` subclass), but forgot to add the extra cleanup.